main.cpp

```diff
@@ -1,66 +1,76 @@
+{
+    SCB->CCR |= SCB_CCR_UNALIGN_TRP_Msk;
+    uint32_t inv_addr = 0xAAA3;
+    uint32_t val = *(uint32_t *)inv_addr;
+    printf("val= %X\n", (unsigned int)val);
+    return;
+}
 
-#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
-void mbed_error_reboot_callback(mbed_error_ctx *error_context) {
-    printf("\nmbed_error_reboot_callback: System rebooting, reboot error callback received");
-    reboot_error_happened = 1;
-    mbed_get_reboot_error_info(&error_ctx);
+//   Implementation for Mbed OS fatal error reboot callback
+void mbed_error_reboot_callback(mbed_error_ctx *error_context)
+{
+    reboot_error_happened = true;
+    err_status = error_context->error_status;
+    printf("\nError:\n");
+    printf("Status: 0x%lX\n", (uint32_t)error_context->error_status);
+    printf("Value: 0x%lX\n", (uint32_t)error_context->error_value);
+    printf("Address: 0x%lX\n", (uint32_t)error_context->error_address);
+    printf("Reboot count: 0x%lX\n", (uint32_t)error_context->error_reboot_count);
+    printf("CRC: 0x%lX\n", (uint32_t)error_context->crc_error_ctx);
     mbed_reset_reboot_error_info();
 }
 #endif
 
 // main() runs in its own thread in the OS
-int main() {
-#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED  
-    printf("\n\nMbed-OS crash reporting test: main()\n");
-    if((reboot_error_happened == 0) || continuous_reboot_test) {
-        printf("\nForcing exception\n");
+int main()
+{
+#if MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED
+    printf("This is the crash reporting Mbed OS example\n");
+    if (!reboot_error_happened) {
+        // Inject the fault exception
         generate_bus_fault_unaligned_access();
-        printf("\nForcing exception failed\n");
     } else {
-        if(error_ctx.error_status < 0) {
-            printf("\nSuccessfully read reboot info ==> \n");
-            printf("\n  error status: 0x%08lX", (uint32_t)error_ctx.error_status);
-            printf("\n  error value: 0x%08lX", (uint32_t)error_ctx.error_value);
-            printf("\n  error address: 0x%08lX", (uint32_t)error_ctx.error_address);
-            printf("\n  error reboot count: 0x%08lX", (uint32_t)error_ctx.error_reboot_count);
-            printf("\n  error crc: 0x%08lX\n", (uint32_t)error_ctx.crc_error_ctx);
-            
-            //Read fault context
-            if(error_ctx.error_status == MBED_ERROR_HARDFAULT_EXCEPTION) {
+        if (err_status < 0) {            
+            if (err_status == MBED_ERROR_HARDFAULT_EXCEPTION) {
+                // Read the fault context
                 mbed_get_reboot_fault_context(&fault_ctx);
-                            
-                printf("\nCrash Report data captured:\n");
-                for(int i=0;i<16;i++) {
-                    printf("[%d]: 0x%08X\n", i, ((uint32_t *)(&fault_ctx))[i]);
+
+                // Print the fault context
+                for (int i = 0; i < 16; i++) {
+                    printf("%d: 0x%X\n", i, ((unsigned int *)(&fault_ctx))[i]);
```

Comment:
You could use the text from `print_context_info` to provide more information on the register. see 
```
//Context Regs
    for (int i = 0; i < 13; i++) {
        mbed_error_printf("\nR%-4d: %08" PRIX32, i, ((uint32_t *)(mbed_fault_context))[i]);
    }

    mbed_error_printf("\nSP   : %08" PRIX32
                      "\nLR   : %08" PRIX32
                      "\nPC   : %08" PRIX32
                      "\nxPSR : %08" PRIX32
                      "\nPSP  : %08" PRIX32
                      "\nMSP  : %08" PRIX32, mbed_fault_context->SP_reg, mbed_fault_context->LR_reg, mbed_fault_context->PC_reg,
                      mbed_fault_context->xPSR, mbed_fault_context->PSP, mbed_fault_context->MSP);
```